File: C++/Hard/RuntimeError/k-inverse-pairs-array_0.cpp
class Solution {
public:
    int kInversePairs(int n, int k) {
        
        if (k > n * (n - 1) / 2 || k < 0) return 0;
        if (k == 0 || k == n * (n - 1) / 2) return 1;
        
        vector<vector<int>> dp(n + 1, vector<int>(k + 1, 0));
        dp[2][0] = 1;
        dp[2][1] = 1;
        for (int i = 3; i <= n; ++i) {
            dp[i][0] = 1;
            for (int j = 1; j <= k; ++j) {
                dp[i][j] = dp[i][j - 1] + dp[i - 1][j];
                if (j >= i) dp[i][j] -= dp[i - 1][j - i];
                dp[i][j] = (dp[i][j] + 1000000007) % 1000000007;
            }
        }
        return dp[n][k];
    }
};

File: Rust/Hard/WrongAnswer/k-inverse-pairs-array_0.rs
impl Solution {
    pub fn k_inverse_pairs(n: i32, k: i32) -> i32 {
        
        let n = n as usize;
        let k = k as usize;
        let mut dp = vec![vec![0; k + 1]; n + 1];
        dp[0][0] = 1;
        for i in 1..=n {
            dp[i][0] = 1;
            for j in 1..=k {
                dp[i][j] = dp[i][j - 1] + dp[i - 1][j];
                if j >= i {
                    dp[i][j] -= dp[i - 1][j - i];
                }
                dp[i][j] = (dp[i][j] + 1000000007) % 1000000007;
            }
        }
        dp[n][k] as i32
    }
}